src/main/java/pl/palonek/interview/course/CourseConstroller.java

```diff
@@ -0,0 +1,44 @@
+package pl.palonek.interview.course;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import pl.palonek.interview.course.model.CourseSessionDTO;
+import pl.palonek.interview.course.model.CourseSummary;
+import pl.palonek.interview.course.service.CourseStatusService;
+import pl.palonek.interview.course.service.CourseSummaryService;
+
+@RestController
+@Slf4j
+@RequiredArgsConstructor
+public class CourseConstroller {
+
+    private final CourseStatusService courseStatusService;
+
+    private final CourseSummaryService courseSummaryService;
+
+    @PostMapping("/status/course")
+    public void postCourseStatus(@RequestBody CourseSessionDTO courseSessionDTO) {
+        log.info("Saving session status for user {} and course {}", courseSessionDTO.getUserId(), courseSessionDTO.getCourseId());
+        courseStatusService.saveCourseStatus(courseSessionDTO);
```

Comment:
[major/minor] Controller passes not validated object into services. It's a major issue on the one side, but minor on the other because it's easy to implement.

[minor] CourseStatusService::saveCourseStatus should return an object, which would inform the Controller what was the result of operation, so that controller can return HTTP OK/Created/Accepted or 404 codes based on it.